pkg/codec/x264/bridge.h

```diff
@@ -0,0 +1,94 @@
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <x264.h>
+
+#define ERR_DEFAULT_PRESET -1
+#define ERR_APPLY_PROFILE -2
+#define ERR_ALLOC_PICTURE -3
+#define ERR_OPEN_ENGINE -4
+#define ERR_ENCODE -5
+
+typedef struct Slice {
+  unsigned char *data;
+  int data_len;
+} Slice;
+
+typedef struct Encoder {
+  x264_t *h;
+  x264_picture_t pic_in, pic_out;
+  x264_param_t param;
+} Encoder;
+
+Encoder *enc_new(x264_param_t param, int* rc) {
+  Encoder *e = (Encoder *)malloc(sizeof(Encoder));
+
+  if (x264_param_default_preset(&e->param, "veryfast", "zerolatency") < 0) {
+    *rc = ERR_DEFAULT_PRESET;
+    goto fail;
+  }
+
+  /* Configure non-default params */
+  e->param.i_csp = param.i_csp;
+  e->param.i_width = param.i_width;
+  e->param.i_height = param.i_height;
+  e->param.i_fps_num = param.i_fps_num;
+  e->param.i_fps_den = 1;
+  // Intra refres:
+  e->param.i_keyint_max = param.i_keyint_max;
+  // Rate control:
+  e->param.rc.i_rc_method = X264_RC_CRF;
```

Comment:
I think VBR mode is better especially for WebRTC.
Bitrate of CRF depends on the input video and takes high bitrate if entropy of the input is large.